app/helpers/controllers/saveOrderItem.js

```diff
@@ -0,0 +1,45 @@
+import { logger } from '../../logger';
+import { postOrderItem } from '../api/ordapi/postOrderItem';
+import { putOrderItem } from '../api/ordapi/putOrderItem';
+
+export const saveOrderItem = async ({
+  orderId,
+  orderItemId,
+  orderItemType,
+  accessToken,
+  serviceRecipientId,
+  serviceRecipientName,
+  itemId,
+  itemName,
+  selectedPrice,
+  formData,
+}) => {
+  try {
+    const response = orderItemId === 'neworderitem'
+      ? await postOrderItem({
+        accessToken,
+        orderId,
+        orderItemType,
+        serviceRecipientId,
+        serviceRecipientName,
+        itemId,
+        itemName,
+        selectedPrice,
+        formData,
+      })
+      : await putOrderItem({
+        accessToken,
+        orderId,
+        orderItemId,
+        formData,
+      });
+    return response;
+  } catch (err) {
+    if (err.response.status === 400 && err.response.data && err.response.data.errors) {
+      logger.info(`Validation errors returned from the API ${JSON.stringify(err.response.data.errors)}`);
+      return err.response.data;
+    }
+    logger.error(`Error saving order item for ${itemName} and ${serviceRecipientName} for order id: ${orderId}`);
+    throw new Error();
```

Comment:
The error makes it all the way out of the routes into the generic error handler. Wouldn't it be better to put the error message in the error class, so that the logging, etc gets handled by the generic handler?

Reply:
We also do that in `app/helpers/routes/routerHelper.js` and the `withCatch` function. The generic error handler there will print the full stack trace. This line here gives a specific log error message for this context. 

We can make the errorHandler do a better job at perhaps pass this message through so the generic handler can log it there. But that will involve bit of re-work across the app to do that.

Reply:
It would be better to combine the error message with the stack trace, so they appear together in a single message. If the system is under any load / once we have splunk in place, identifying the association of adjacent messages will be a lot harder. Can you please add a tech debt ticket to look at it?

Reply:
tech debt ticket #8537 created